pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -308,7 +310,24 @@ func (r *ReconcileClusterDeployment) Reconcile(request reconcile.Request) (recon
 		cdLog.Errorf("error getting job: %v", err)
 		return reconcile.Result{}, err
 	} else {
+		// If install job exists but the cd generation is different, recreate the job after deleting it
+		// provided the cd is not installed
 		cdLog.Infof("cluster job exists, successful: %v", cd.Status.Installed)
+		cdLog.Infof("cluster job exists, generation: %v", job.Annotations[clusterDeploymentGeneration])
+		if !cd.Status.Installed {
+			cdGeneration := strconv.FormatInt(cd.Generation, 10)
+			cdLog.Infof("cluster generation: %v", cdGeneration)
+			if existingJob.Annotations[clusterDeploymentGeneration] != cdGeneration {
```

Comment:
Lets get some unit tests added here, and include one that covers this case above where the job exists but has *no* cluster deployment generation set in it's annotations.

Reply:
So if you look at generate.go, where we generate our installer job, I have written code for always adding a generation to the annotation when we generate an installer job. There I also have a check so that if Annotations are null I insert a map of strings (key, value both strings)

Reply:
That is if you generate a new job, but we're talking about an environment with pre-existing jobs. AFAIK if the job already exists, it won't be regenerated and won't pickup the new annotations. Am I mistaken on this?

Reply:
Ahhh, I get it now. Perfect, makes sense.